source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1153,16 +1153,18 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::connPool(
 
   // Note: to simplify this, we assume that the factory is only called in the scope of this
   // function. Otherwise, we'd need to capture a few of these variables by value.
-  ConnPoolsContainer::ConnPools::OptPoolRef pool = container.pools_->getPool(hash_key, [&]() {
-    return parent_.parent_.factory_.allocateConnPool(
-        parent_.thread_local_dispatcher_, host, priority, protocol,
-        have_options ? context->downstreamConnection()->socketOptions() : nullptr);
-  });
-  // The Connection Pool tracking is a work in progress. We plan for it to eventually have the
-  // ability to fail, but until we add upper layer handling for failures, it should not. So, assert
-  // that we don't accidentally add conditions that could allow it to fail.
-  ASSERT(pool.has_value(), "Pool allocation should never fail");
-  return &(pool.value().get());
+  ConnPoolsContainer::ConnPools::OptPoolRef pool =
```

Comment:
If you look at line 1135, you'll note that the priority is set as part of the key. I don't think this is necessary any longer. Shall I remove it?

Reply:
Sounds like by passing in both `hash_key` and `priority` you're already partitioning by priority so hashing on priority seems unnecessary? If so yeah you can probably remove it

Reply:
Looks like there's some UT which shows that the behaviour works. With the PriorityConnPool working, the ClusterManagerImplTest passes. With it hard-coding the priority to default (I.e. ignoring the input), these tests fail:

```
[  FAILED  ] ClusterManagerImplTest.CloseHttpConnectionsOnHealthFailure
[  FAILED  ] ClusterManagerImplTest.DynamicHostRemove
[  FAILED  ] ClusterManagerImplTest.DynamicHostRemoveWithTls
```

So, that gives some confidence that removing it is correct.